### PR TITLE
Improve auth decorators

### PR DESCRIPTION
**Pull request by raymondfeng**
This PR improves `@authenticate` (implements https://github.com/strongloop/loopback-next/issues/2460)

1. Allow `@authenticate` to be applied at class level
2. Add `@authenticate.skip` to skip authentication

It also improves `@authorize`:

1. Add `@authorize.skip` to skip authorization

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/decorators/Decorators_authenticate.md [hunk 1]
```diff
@@ -40,6 +40,31 @@ export class WhoAmIController {
 }
 ```
 
+Please note that `@authenticate` can also be applied at class level for all
```
Comment (dhmlau) [inline docs/site/decorators/Decorators_authenticate.md]: maybe `applied at class level` -> `applied at the class level`?

File: docs/site/decorators/Decorators_authenticate.md [hunk 2]
```diff
@@ -40,6 +41,33 @@ export class WhoAmIController {
...
+```ts
+@authenticate('BasicStrategy')
+export class WhoAmIController {
+  constructor(
+    @inject(AuthenticationBindings.CURRENT_USER) private user: UserProfile,
+  ) {}
+
+  @get('/whoami')
+  whoAmI(): string {
+    return this.user.id;
```
Comment (jannyHou) [inline docs/site/decorators/Decorators_authenticate.md]: should be `this.user[securityId]` ?

File: docs/site/decorators/Decorators_authenticate.md [hunk 3]
```diff
@@ -40,6 +41,33 @@ export class WhoAmIController {
 }
 ```
 
+To configure a default authentication for all methods within a class,
```
Comment (jannyHou) [inline docs/site/decorators/Decorators_authenticate.md]: 👍 👏 The design looks reasonable to me.

File: packages/authentication/src/decorators/authenticate.decorator.ts [hunk 4]
```diff
@@ -4,50 +4,102 @@
...
+        spec,
+        {decoratorName: '@authenticate'},
+      )(target, method, methodDescriptor);
+    }
+    if (typeof target === 'function' && !method && !methodDescriptor) {
+      // Class
+      return AuthenticateClassDecoratorFactory.createDecorator(
+        AUTHENTICATION_METADATA_CLASS_KEY,
+        spec,
+        {decoratorName: '@authorize'},
```
Comment (jannyHou) [inline packages/authentication/src/decorators/authenticate.decorator.ts]: `@authenticate` instead of `@authorize`?

Comment (raymondfeng) [reply packages/authentication/src/decorators/authenticate.decorator.ts]: Good catch

**Pull request merged**